core/client/src/test/java/alluxio/client/netty/NettyRemoteBlockReaderTest.java

```diff
@@ -0,0 +1,189 @@
+  private long mBlockId = 4242L;
+  private long mOffset = 1;
+  private long mLength = 10;
+  private long mLockId = 6;
+  private long mSessionId = 3421;
+
+  /**
+   * Set up.
+   */
+  @Before
+  public void before() throws InterruptedException {
+    mNettyRemoteBlockReader = new NettyRemoteBlockReader();
+
+    mBootstrap = Mockito.mock(Bootstrap.class);
+    mClientHandler = new ClientHandler();
+
+    Whitebox.setInternalState(mNettyRemoteBlockReader, "mClientBootstrap", mBootstrap);
+    Whitebox.setInternalState(mNettyRemoteBlockReader, "mHandler", mClientHandler);
+
+    ChannelFuture channelFuture = Mockito.mock(ChannelFuture.class);
+    Channel channel = Mockito.mock(Channel.class);
+    Mockito.when(channel.close()).thenReturn(channelFuture);
+    Mockito.when(channelFuture.sync()).thenReturn(channelFuture);
+    Mockito.when(channelFuture.channel()).thenReturn(channel);
+    Mockito.when(mBootstrap.connect(Mockito.any(SocketAddress.class))).thenReturn(channelFuture);
+  }
+
+  /**
+   * Test case for a valid {@link RPCBlockReadResponse} is received.
+   */
+  @Test
+  public void readRemoteBlockTest() throws IOException {
+    Thread channelThread = new Thread(new ChannelReadThread(
+            RPCResponse.Type.RPC_BLOCK_READ_RESPONSE, RPCResponse.Status.SUCCESS
+    ));
+    channelThread.start();
+
+    ByteBuffer byteBuffer = mNettyRemoteBlockReader.readRemoteBlock(mInetSocketAddress,
+            mBlockId, mOffset, mLength, mLockId, mSessionId);
+
+    Assert.assertEquals(10, byteBuffer.capacity());
+    Assert.assertEquals(10, byteBuffer.limit());
+    byte[] dst = new byte[10];
+    byteBuffer.get(dst);
+    Assert.assertEquals("alluxio", new String(dst).substring(0, 7));
+  }
+
+  /**
+   * Test case for an invalid {@link RPCBlockReadResponse} is received.
+   */
+  @Test(expected = IOException.class)
+  public void readRemoteBlockWithBadStatusTest() throws IOException {
+    Thread channelThread = new Thread(new ChannelReadThread(
+            RPCResponse.Type.RPC_BLOCK_READ_RESPONSE, RPCResponse.Status.BLOCK_LOCK_ERROR
+    ));
+    channelThread.start();
+
+    mNettyRemoteBlockReader.readRemoteBlock(mInetSocketAddress,
+            mBlockId, mOffset, mLength, mLockId, mSessionId);
+  }
+
+  /**
+   * Test case for {@link RPCErrorResponse} is received.
+   */
+  @Test(expected = IOException.class)
+  public void readRemoteBlockErrorResponseTest() throws IOException {
+    Thread channelThread = new Thread(new ChannelReadThread(
+            RPCResponse.Type.RPC_ERROR_RESPONSE, RPCResponse.Status.SUCCESS
+    ));
+    channelThread.start();
+    mNettyRemoteBlockReader.readRemoteBlock(mInetSocketAddress,
+            mBlockId, mOffset, mLength, mLockId, mSessionId);
+  }
+
+  /**
+   * Test case for {@link RPCFileWriteResponse} is received.
+   */
+  @Test(expected = IOException.class)
+  public void readRemoteBlockUnknownResponseTest() throws IOException {
```

Comment:
this is called `readRemoteBlockUnknownResponseTest`, but it tests receiving a `RPCFileWriteResponse`?